### PR TITLE
[lgthinq][WIP] Initial contribution

### PR DESCRIPTION
**Pull request by nemerdaud**
This is the initial contribution for LG Thinq Binding for OH3. 
This binding are composed by a Bridger and Air Conditioned Thing currently. The other Things will come soon.
I'm already using this addon for a while but to respect the contribution guide, I'm still working improving the unit tests and documentations.

### CODE REVIEW

**Comment by hmerk** [comment]
Please sign-off at least you last commit to make DCO check happy.
(git commit -s )

**Comment by hmerk** [comment]
Thanks for your contribution.
Could you please check
https://www.openhab.org/docs/developer/guidelines.html#null-annotations
as you are not censequent and corret using the annotations.

File: bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/api/RestUtils.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+
+import javax.ws.rs.core.UriBuilder;
+
+import org.apache.commons.codec.digest.HmacAlgorithms;
+import org.apache.commons.codec.digest.HmacUtils;
```
Comment (hmerk) [inline bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/api/RestUtils.java:30]: Usage of apache commons is deprecated, so please change.
See https://github.com/openhab/openhab-addons/issues/7722

File: bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/handler/LGBridgeHandler.java [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LGBridgeHandler}
+ *
+ * @author Nemer Daud - Initial contribution
+ */
+public class LGBridgeHandler extends ConfigStatusBridgeHandler implements LGBridge {
```
Comment (hmerk) [inline bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/handler/LGBridgeHandler.java:50]: Should be annotated as @NonNullByDefault, please check for all classes

File: bundles/org.openhab.binding.lgthinq/src/main/resources/OH-INF/i18n/lgthinq_xx.properties [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# FIXME: please substitute the xx with a proper locale, ie. de
```
Comment (hmerk) [inline bundles/org.openhab.binding.lgthinq/src/main/resources/OH-INF/i18n/lgthinq_xx.properties:1]: Please note this comment and rename lgthinq_xx.properties to lgthinq.properties.

File: bundles/org.openhab.binding.lgthinq/src/test/java/org/openhab/binding/lgthinq/handler/LGBridgeTests.java [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+    // // });
+    // // service1.changeOperationMode("d27cc560-7149-11d3-80b6-7440bec3653e", ACOpMode.FAN);
+    // System.out.println("AC power ON:");
+    // } catch (LGThinqException e) {
+    // logger.error("Error testing facade", e);
+    // // } catch (InterruptedException | IOException e) {
+    // // e.printStackTrace();
+    // }
+    // }
+
```
Comment (hmerk) [inline bundles/org.openhab.binding.lgthinq/src/test/java/org/openhab/binding/lgthinq/handler/LGBridgeTests.java:322]: Please remove unused comments

**Review by hmerk — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/api/RestUtils.java (hunk 1 above)
Comment (nemerdaud) [reply bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/api/RestUtils.java:30]: Done !

File: bundles/org.openhab.binding.lgthinq/src/main/resources/OH-INF/i18n/lgthinq_xx.properties (hunk 3 above)
Comment (nemerdaud) [reply bundles/org.openhab.binding.lgthinq/src/main/resources/OH-INF/i18n/lgthinq_xx.properties:1]: Sorry. Shame on me :-(

File: bundles/org.openhab.binding.lgthinq/src/test/java/org/openhab/binding/lgthinq/handler/LGBridgeTests.java (hunk 4 above)
Comment (nemerdaud) [reply bundles/org.openhab.binding.lgthinq/src/test/java/org/openhab/binding/lgthinq/handler/LGBridgeTests.java:322]: Done. I'm still changing the test to cover better the code. But I will keep it clean before update the PR's.

**Comment by hmerk** [comment]
Now you have borked your PR by adding commits not related to your binding.
Please rebase your lokal repo/branch, signoff and push.
Another general advise, THING_TYPE constants have to be defined in BindingConstants and imported from there. Also please check your folder structure, which seems not to be correct.

**Review by hmerk — COMMENTED** [review]
srarted by mistake

**Comment by hmerk** [comment]
Still not okay and including unrelated PRs.

**Comment by nemerdaud** [comment]
I know, @hmerk. I'm not getting to work the rebase of the branch. There are a lot of conflicts... It's a mess. I did a wrong pull from main in this branch by mistake, and push without pay attention on it. I don't know... I think the fast and better to save time is to cancel this PR, then I can create a new fresh branch, update with my changes and push (signed-off) again in a new PR. What do you think ?

**Comment by hmerk** [comment]
Seems to be the best idea. I had similar issues in the past….

**Pull request closed without merging**